syntax = "proto2";

package e2e_noa.planning;
  

message PlannerStatusProto {
  enum PlannerStatusCode {
    OK = 1;
    ROUTE_MSG_UNAVAILABLE = 2;
    INPUT_INCORRECT = 3;
    ROUTING_FAILED = 4;
    OBJECT_MANAGER_FAILED = 5;
    REFERENCE_PATH_UNAVAILABLE = 6;
    SCHEDULER_UNAVAILABLE = 7;
    DECISION_CONSTRAINTS_UNAVAILABLE = 8;
    INITIALIZER_FAILED = 9;
    OPTIMIZER_FAILED = 10;
    PATH_EXTENSION_FAILED = 11;
    SPEED_OPTIMIZER_FAILED = 12;
    TRAJECTORY_VALIDATION_FAILED = 13;
    FALLBACK_PREVIOUS_INFO_UNAVAILABLE = 14;
    START_POINT_PROJECTION_TO_ROUTE_FAILED = 15;
    SELECTOR_FAILED = 16;
    RESET_PREV_TARGET_LANE_PATH_FAILED = 17;
    TRAFFIC_LIGHT_INFO_COLLECTOR_FAILED = 20;
    PLANNER_STATE_INCOMPLETE = 21;
    PATH_PLANNER_FAILED = 22;
    TARGET_LANE_CANDIDATES_UNAVAILABLE = 23;
    MODEL_PRECONDITION_CHECK_FAILED = 24;
    MODEL_INFERENCE_FAILED = 25;
    MODEL_OUTPUT_POSTPROCESS_FAILED = 26;
    UPDATE_TARGET_LANE_PATH_FAILED = 27;
    LOCAL_LANE_MAP_BUILDER_FAILED = 28;
    LCC_ODC_CHECK_FAIL = 29;
    BUILD_DRIVING_MAP_FAILED = 30;
    ACC_CORRIDOR_FAILED = 31;
    ACC_SPEED_FAILED = 32;
    ACC_TRAJECTORY_VALIDATION_FAILED = 33;
    BRANCH_RESULT_IGNORED = 34;
    LC_SAFETY_CHECK_FAILED = 35;
    GEAR_NOT_READY = 36;
    PROJECT_TO_ONLINE_MAP_FAILED = 37;
    LC_LINE_CHECK_FAILED = 38;
    LC_MULTI_LANES_FAILES = 39;
    LC_MISS_NAVI_FAILED = 40;
    LC_DEVIATE_NAVI = 41;
    LC_SPECIAL_LANE_CHECK_FAILED = 42;
    PLANNER_ABNORMAL_EXIT = 99;
    EXPERT_TRAJ_UNAVAILABLE = 200;
    EXPERT_TRAJ_INTERMEDIATES_RECONSTRUCTION_FAILED = 201;
    EXPERT_SPEED_IN_CONTRADICTORY_WITH_SPEED_FINDER = 202;
    TARGET_LANESEQUENCE_SHORTLY = 210;
    NOT_FIND_NEAREST_LANE = 211;
    OPTIMIZER_DIFF_INTENTION_EXPERT = 230;
    TARGET_LANE_JUMPED_FAIL = 231;
    TRAJECTORY_NOT_ENTERING_TARGET_LANE = 235;
    FRONT_DANGEROUS_VEHICLE_TARGET_LANE = 236;
    REAR_DANGEROUS_VEHICLE_TARGET_LANE = 237;
    FRONT_DANGEROUS_VEHICLE_CURRENT_LANE = 238;
    REAR_DANGEROUS_VEHICLE_CURRENT_LANE = 239;
    DANGEROUS_STATIONARY_OBSTACLE = 240;
    DEVIATE_NAVI = 255;
    reserved 18, 19;
  }
  optional PlannerStatusCode status = 1;
  optional string message = 2;
}
